enterprise/coderd/licenses.go

```diff
@@ -0,0 +1,181 @@
+package coderd
+
+import (
+	"context"
+	"crypto/ed25519"
+	_ "embed"
+	"net/http"
+	"time"
+
+	"golang.org/x/xerrors"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/golang-jwt/jwt/v4"
+
+	"cdr.dev/slog"
+
+	"github.com/coder/coder/coderd"
+	"github.com/coder/coder/coderd/database"
+	"github.com/coder/coder/coderd/httpapi"
+	"github.com/coder/coder/coderd/rbac"
+	"github.com/coder/coder/codersdk"
+)
+
+const (
+	CurrentVersion        = 3
+	HeaderKeyID           = "kid"
+	AccountTypeSalesforce = "salesforce"
+	VersionClaim          = "version"
+	PubSubEventLicenses   = "licenses"
+)
+
+var ValidMethods = []string{"EdDSA"}
+
+//go:embed keys/2022-08-12
+var key20220812 []byte
```

Comment:
It's not obvious to me why we have to embed a static key. A comment here would be helpful!